ARCHITECTURE.md

```diff
@@ -26,8 +30,41 @@ The graphql server can scale horizontally with additional pods.
 
 Redis is being used as a distributed lock such that several accounts can't be modified at the same time by different instances. 
 
-### Database
+## Database
+
+The state of the wallet is shared among 4 differences sources: 
+
+### bitcoin-core
+
+bitcoin core handle the cold storage. 
+
+there is typically a very low load transaction wise in regard to the wallet. maybe 1 transaction per week when a rebalancing is needed.
+
+### lnd 
+
+lnd handle all of the transactions the users run.
+lnd stores itself the data in a bbolt database.
+
+every succesful payment within lnd is being recorded on mongodb.
+
+offchain-wise, nothing is needed within the lnd side to present a transaction to a user after it has been executed.
+onchain-wise, it's possible that a user shared an onchain address. this is currently tied to lnd. (we intend to not use lnd for onchain transaction to remove this dependancy and simplify lnd instance recyling)
+
+### Mongodb
+
+Mongodb is the database storing all users transactions. It's the source of truth for the customers' accounts.
+
+To achieve strong consistency, it's currently setup with:
+- Write access need to be valided by 2-out-of-3 pods before being considered succesful
+- Schema on write with mongoose. 
+
+The accounting part is being done by [medici](https://github.com/flash-oss/medici/commits/master).
+
+There is no transaction used yet (the main inconsistency if this were to happen would be between the journal entries and the transaction entries). A [contribution on this](https://github.com/flash-oss/medici/issues/23) is welcome. It has not been a focus yet because there hasn't been any issue on this yet. 
+
+### Redis
 
-Mongodb is the main database which is storing the transactions. It's the source of truth for the customers' accounts.
+Redis is used a distributed lock, such that the number of replicas for the node/graphql can be scaled as needed.
+Redis is also used for query rate limiting , and to store ephemeral data such as result of route probing.
 
-It's currently setup by default with auto replication on a 2-out-of-3 setup, and writes are only accepted as successful when a majority (2) of the nodes have 
+Redis can be though as a cache. The data can be deleted as needed (it would reset the lock, so some attention needed to be paid on this if some live queries are running)
```

Comment:
Redis is explicit about not providing any consistency guarantees:

>> Redis Cluster is not able to guarantee strong consistency. In practical terms this means that under certain conditions it is possible that Redis Cluster will lose writes that were acknowledged by the system to the client.
https://redis.io/topics/cluster-tutorial

Using it as a distributed lock doesn't seem a viable solution when running at scale. 2 concurrent attempts to acquire a lock may both complete successfully with unknown consequences.

Reply:
Actually the write side also has problems. We are using `majority` concern but not specifying the `j` option to only acknowledge a write when the operation has persisted to disk https://docs.mongodb.com/manual/reference/write-concern/ is set to false.

Hence we don't have consistent writes either... just in memory.

Reply:
this is the page to look at for the lock we are using: https://redis.io/topics/distlock